### ISSUE TITLE
EVM property doesn't contain event emitter properties

### ISSUE THREAD

**Issue by acolytec3** [ethereumjs/ethereumjs-monorepo#2181]
As identified by the [Ganache team](https://github.com/ethereumjs/ethereumjs-monorepo/pull/2159#issuecomment-1219658752), the current `vm.evm` property is typed as `EVMInterface | EVM` and this leads to DX challenges where 
```js
const vm = await VM.create({})
vm.evm.on("step"...  //yields a type error 
```
yields a type error in the IDE because Typescript can't infer the type `EVM` for `vm.evm` due to the `EVMInterface` not extending the `EventEmitter` class.  For users that want to access the `step` event and other the other public properties/events of the `EVM` class while referencing it from the VM, they are currently forced to do:
```js
;(vm.evm as EVM).on...
```
each time they want to access something exposed by the `EVM` class but not included in the interface.  The "Typescript Way" :tm: of handling stuff like this is through a [type guard](https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates),  I propose adding a type guard as a static method on the EVM class to allow teams to quickly access the type of the EVM  property once at the start of their code and then not have to worry about typecasting.  

It would look something like this:
```js

class EVM extends EventEmitter {
// class implementation
const static isEVM = (evm: EVMInterface | EVM): evm is EVM => {
  return (evm as EVM)._transientStorage !== undefined
// more class implementation stuff
}
```

and be used something like:
```js
import { EVM } from '@ethereumjs/evm'
import { VM } from '@ethereumjs/vm'

tape('test vm', async (t) => {
  const vm = await VM.create({})

  if (!EVM.isEVM(vm.evm)) throw new Error("EVM must be a real EVM!!!")

  vm.evm.on... //not a type error!
})
```

WDYGT?

cc: @davidmurdoch @holgerd77 @jochem-brouwer

**Comment by davidmurdoch** [comment]
Why not make the `EVMInterface` match the actual implementation?

**Comment by jochem-brouwer** [comment]
If we do that, we require the interface to implement those events, which is not exactly necessary to work with our libraries.

For a non-runtime fix see #2182

**Comment by davidmurdoch** [comment]
Your Generics solution looks to be more useful to Ganache!

**Comment by jochem-brouwer** [comment]
That's great, in that case I will continue on that tomorrow (unless @acolytec3 or someone else has reservations here). We can most likely also solve the problem with the `EEIInterface`.

**Comment by jochem-brouwer** [comment]
This is closed by #1955, #2184, #2235
